compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -871,6 +872,45 @@ public void visit(BLangSimpleVariable varNode) {
         transferForkFlag(varNode);
     }
 
+    private void analyzeModuleConfigurableAmbiguity(BLangPackage pkgNode) {
+        if (pkgNode.moduleContextDataHolder == null) {
+            return;
+        }
+        ModuleDescriptor rootModule = pkgNode.moduleContextDataHolder.descriptor();
+        Map<String, PackageID> configKeys = new HashMap<>();
+        Set<BVarSymbol> configVars = symResolver.getConfigVarSymbols(pkgNode.symbol);
+        populateModuleKeys(configVars, rootModule, configKeys);
+        for (BVarSymbol variable : configVars) {
+            validateMapConfigVariable(variable.pkgID.orgName.value + "." + variable.pkgID.name.value + "."
+                    + variable.name.value, variable, configKeys);
+            if (variable.pkgID.getOrgName().value.equals(rootModule.org().value())) {
+                validateMapConfigVariable(variable.pkgID.name.value + "." + variable.name.value, variable, configKeys);
+                if (variable.pkgID.getName().value.equals(rootModule.packageName().value()) &&
+                        !(variable.name.equals(variable.pkgID.name))) {
+                    validateMapConfigVariable(variable.name.value, variable, configKeys);
+                }
+            }
+        }
+    }
+
+    private void populateModuleKeys(Set<BVarSymbol> configVars, ModuleDescriptor rootModule,
+                                    Map<String, PackageID> configKeys) {
+        for (BVarSymbol variable : configVars) {
+            configKeys.put(variable.pkgID.orgName.value + "." + variable.pkgID.name.value, variable.pkgID);
+            if (!variable.pkgID.getOrgName().value.equals(rootModule.org().value())) {
```

Comment:
`rootModule.org().value()` can be extracted out to a variable outside the for loop. Better yet, the org name can be accepted as the param, instead of `ModuleDescriptor`.